src/urllib3/connection.py

```diff
@@ -198,22 +258,41 @@ def putheader(self, header, *values):
                 )
             )
 
-    def request(self, method, url, body=None, headers=None):
+    def request(
+        self,
+        method: str,
+        url: str,
+        body: Optional[_DefaultType] = None,
+        headers: Optional[Mapping[str, str]] = None,
+        *,
+        encode_chunked: bool = False,
+    ) -> None:
         if headers is None:
             headers = {}
         else:
             # Avoid modifying the headers passed into .request()
-            headers = headers.copy()
+            headers = copy(headers)
```

Comment:
Can we avoid using `copy.copy()`? Why does mypy balk at `headers = headers.copy()`?

Reply:
This is because `headers` are of type `Mapping[str, str]` and `Mapping` does not have a `copy` method. Changing `headers` type would again violate LSP, so the same situation as with the `code_enchunked` discussion. So, probably makes sense to add comments and type ignores again and make headers to be `dict` for now? This also raises the question if in the `BaseHTTPConnection.request` should take headers as `dict` or `HTTPHeaderDict` (or union of both, which seems best I think). Related to #2158